File: webapp/main/src/store/actions/ServiceModel.ts
import { LocalDate } from "../viewcontext/TimePeriod";

export interface ServiceAppModel extends EntityId {
  servicemanEmail: string | undefined,
  servicemanName: string | undefined,
  description: string | undefined,
  when: LocalDate,
  duration: number | undefined,
  distance: number | undefined
  customer: {
    id: EntityId,
    name: string
  } | undefined
}

export interface EntityId {
  projectId: string,
  entityId: string,
  entityVersion: number,
}

File: uservice-webapi/main/host/src/main/java/sinnet/ServiceModel.java
package sinnet;

import java.time.LocalDate;

import org.eclipse.microprofile.graphql.Id;
import org.eclipse.microprofile.graphql.NonNull;

import lombok.Data;

@Data
public class ServiceModel {
  private @NonNull @Id String projectId;
  private @NonNull @Id String entityId;
  private @NonNull int entityVersion;
  private CustomerEntity customer;
  private String servicemanName;
  private @NonNull LocalDate whenProvided;
  private String description;
  private Integer distance;
  private Integer duration;
}
